janusgraph-core/src/main/java/org/janusgraph/graphdb/vertices/CacheVertex.java

```diff
@@ -38,6 +38,10 @@ public CacheVertex(StandardJanusGraphTx tx, long id, byte lifecycle) {
         queryCache = new HashMap<>(4);
     }
 
+    public void refresh() {
+        queryCache.clear();
+    }
```

Comment:
I see that all operations with queryCache are synchronized except of this method. I assume this method should be thread safe as well. Thus, I guess it should be:
```
synchronized (queryCache) {
  queryCache.clear();
}
```